cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -1106,23 +1108,31 @@ private V1Container createContainer(String name, String containerImage, String i
    * Creates a {@link V1ResourceRequirements} based on the given {@link ResourceSpecification}. If the namespace has a
    * resource quota, the objects must also specify resource limits.
    */
-  private V1ResourceRequirements createResourceRequirements(ResourceSpecification resourceSpec) {
+  private V1ResourceRequirements createResourceRequirements(Type resourceType, ResourceSpecification resourceSpec) {
     Map<String, String> cConf = masterEnvContext.getConfigurations();
     float cpuMultiplier = Float.parseFloat(cConf.getOrDefault(CPU_MULTIPLIER, DEFAULT_MULTIPLIER));
     float memoryMultiplier = Float.parseFloat(cConf.getOrDefault(MEMORY_MULTIPLIER, DEFAULT_MULTIPLIER));
+
+    if (resourceType == V1Job.class) {
+      // For job type use program multipliers.
```

Comment:
why just for jobs? Is this because we're trying to differentiate between user programs and system pods?

Reply:
yea thats because we are using separate multiplier for user programs.

Reply:
is there a better way to check that it's a user program? Technically somebody can still try and run a service as a user program, or somebody could write a system app that runs spark

Reply:
Hmm, currently we dont distinguish between user program and system program on kube environment side as we directly interact with Twill apis. But I think its good to add that distinction.

Reply:
Filed this: https://cdap.atlassian.net/browse/CDAP-19218